zipkin-server/src/main/java/zipkin/server/ZipkinServer.java

```diff
@@ -13,17 +13,70 @@
  */
 package zipkin.server;
 
+import com.nike.riposte.server.Server;
+import com.nike.riposte.server.config.ServerConfig;
+import com.nike.riposte.server.http.Endpoint;
+import com.nike.riposte.server.http.filter.RequestAndResponseFilter;
+import com.nike.riposte.server.http.filter.impl.AllowAllTheThingsCORSFilter;
+
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 import zipkin.server.brave.BootstrapTrace;
 
+import static java.util.Collections.singletonList;
+
 @SpringBootApplication
 @EnableZipkinServer
 public class ZipkinServer {
 
-  public static void main(String[] args) {
-    new SpringApplicationBuilder(ZipkinServer.class)
+  public static void main(String[] args) throws CertificateException, InterruptedException, IOException {
+    ConfigurableApplicationContext cac = new SpringApplicationBuilder(ZipkinServer.class)
         .listeners(new RegisterZipkinHealthIndicators(), BootstrapTrace.INSTANCE::record)
         .properties("spring.config.name=zipkin-server").run(args);
+
+//    AnnotationConfigApplicationContext rootContext = new AnnotationConfigApplicationContext(ZipkinServer.class);
+//    rootContext.addApplicationListener(new RegisterZipkinHealthIndicators());
+//    rootContext.addApplicationListener(BootstrapTrace.INSTANCE::record);
+
+    Server server = new Server(new RiposteServerConfig(cac));
+    server.startup();
   }
+
+  public static class RiposteServerConfig implements ServerConfig {
+    private final Collection<Endpoint<?>> endpoints;
+    private final List<RequestAndResponseFilter> requestAndResponseFilters =
+      singletonList(new AllowAllTheThingsCORSFilter());
```

Comment:
good point. we don't have a way of saying "no-cors filtering" I think.. @shakuzen correct me if I'm wrong